index.js

```diff
@@ -3,15 +3,22 @@ var jwt = require('jsonwebtoken');
 var request = require('request');
 var nodeCache = require('node-cache');
 var parseCacheControl = require('parse-cache-control');
+var hash = require('string-hash');
 
 var credential_cache = new nodeCache({ useClones: false });
+var hashCacheKey = false;
 
 var logger = console.log;
 
 var REFRESH_TOKEN_CLIENT_ID = process.env.DEFAULT_TARGET_ID || 'QkxOvNz4fWRFT6vcq79ylcIuolFz2cwN';
 
 var construct_cache_key = function (method, url, authToken) {
-  return "" + method + "-" + url + "-" + authToken;
+  var decodedToken = jwt.parse(authToken);
+  if (hashCacheKey) {
+    return hash("" + method + "-" + url + "-" + decodedToken.sub);
```

Comment:
How does this behave in the case of a hash collision? From what I'm reading it looks like we'd expose the other user's data.